File: src/main/java/com/abuabdul/mytravelpal/data/document/MyTravelPal.java
/*
 * Copyright 2013-2016 abuabdul.com
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * 
 */
package com.abuabdul.mytravelpal.data.document;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 * @author abuabdul
 *
 */
@Document(collection = MyTravelPal.MYTRAVELPAL_COLL_NAME)
public class MyTravelPal {

	public static final String MYTRAVELPAL_COLL_NAME = "MyTravelPlans";

	@Id
	private String id;

	private String travelPlanDesc;
	private String startDate;
	private String startTime;
	private String endDate;
	private String endTime;
	private String travelMode;
	private String travelType;
	private String sideNote;
	private String createdDate;
	private String createdBy;
	private String updatedDate;
	private String updatedBy;

	/**
	 * @return the id
	 */
	public String getId() {
		return id;
	}

	/**
	 * @param id
	 *            the id to set
	 */
	public void setId(String id) {
		this.id = id;
	}

	/**
	 * @return the travelPlanDesc
	 */
	public String getTravelPlanDesc() {
		return travelPlanDesc;
	}

	/**
	 * @param travelPlanDesc
	 *            the travelPlanDesc to set
	 */
	public void setTravelPlanDesc(String travelPlanDesc) {
		this.travelPlanDesc = travelPlanDesc;
	}

	/**
	 * @return the startDate
	 */
	public String getStartDate() {
		return startDate;
	}

	/**
	 * @param startDate
	 *            the startDate to set
	 */
	public void setStartDate(String startDate) {
		this.startDate = startDate;
	}

	/**
	 * @return the startTime
	 */
	public String getStartTime() {
		return startTime;
	}

	/**
	 * @param startTime
	 *            the startTime to set
	 */
	public void setStartTime(String startTime) {
		this.startTime = startTime;
	}

	/**
	 * @return the endDate
	 */
	public String getEndDate() {
		return endDate;
	}

	/**
	 * @param endDate
	 *            the endDate to set
	 */
	public void setEndDate(String endDate) {
		this.endDate = endDate;
	}

	/**
	 * @return the endTime
	 */
	public String getEndTime() {
		return endTime;
	}

	/**
	 * @param endTime
	 *            the endTime to set
	 */
	public void setEndTime(String endTime) {
		this.endTime = endTime;
	}

	/**
	 * @return the travelMode
	 */
	public String getTravelMode() {
		return travelMode;
	}

	/**
	 * @param travelMode
	 *            the travelMode to set
	 */
	public void setTravelMode(String travelMode) {
		this.travelMode = travelMode;
	}

	/**
	 * @return the travelType
	 */
	public String getTravelType() {
		return travelType;
	}

	/**
	 * @param travelType
	 *            the travelType to set
	 */
	public void setTravelType(String travelType) {
		this.travelType = travelType;
	}

	/**
	 * @return the sideNote
	 */
	public String getSideNote() {
		return sideNote;
	}

	/**
	 * @param sideNote
	 *            the sideNote to set
	 */
	public void setSideNote(String sideNote) {
		this.sideNote = sideNote;
	}

	/**
	 * @return the createdDate
	 */
	public String getCreatedDate() {
		return createdDate;
	}

	/**
	 * @param createdDate
	 *            the createdDate to set
	 */
	public void setCreatedDate(String createdDate) {
		this.createdDate = createdDate;
	}

	/**
	 * @return the createdBy
	 */
	public String getCreatedBy() {
		return createdBy;
	}

	/**
	 * @param createdBy
	 *            the createdBy to set
	 */
	public void setCreatedBy(String createdBy) {
		this.createdBy = createdBy;
	}

	/**
	 * @return the updatedDate
	 */
	public String getUpdatedDate() {
		return updatedDate;
	}

	/**
	 * @param updatedDate
	 *            the updatedDate to set
	 */
	public void setUpdatedDate(String updatedDate) {
		this.updatedDate = updatedDate;
	}

	/**
	 * @return the updatedBy
	 */
	public String getUpdatedBy() {
		return updatedBy;
	}

	/**
	 * @param updatedBy
	 *            the updatedBy to set
	 */
	public void setUpdatedBy(String updatedBy) {
		this.updatedBy = updatedBy;
	}

	/**
	 * @return the mytravelpalCollName
	 */
	public static String getMytravelpalCollName() {
		return MYTRAVELPAL_COLL_NAME;
	}
}

File: src/main/webapp/WEB-INF/static-content/js/myTravelPal.js
/*
 * Copyright 2013-2016 abuabdul.com
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * 
 */

/*
 *
 * @project   myTravelPal App
 * @date      15-Dec-2015
 * @author    Abubacker Siddik A, Chennai, India <abuabdul86@hotmail.com>
 * @licensor  Apache License 2.0
 * @site      
 *
 */

//Inline mode
$.fn.editable.defaults.mode = 'inline';
$(function() {

	/* Bootstrap tooltip */
	$('[data-toggle="tooltip"]').tooltip({
		trigger : 'focus'
	});

	/* Bootstrap datatable */
	var travelPlans = $('#TravelPlans').DataTable({
		"columnDefs" : [ {
			"searchable" : false,
			"orderable" : false,
			"targets" : [ 0, 7 ]
		} ],
		"order" : [ [ 1, 'asc' ] ]
	});

	travelPlans.on('order.dt search.dt', function() {
		travelPlans.column(0, {
			search : 'applied',
			order : 'applied'
		}).nodes().each(function(cell, i) {
			cell.innerHTML = i + 1;
		});
	}).draw();

	/* Bootstrap editable */
	$('.editable').editable({
		type : 'text',
		url : baseURL + '/secure/travel/updatePlans.go',
		success : function(response, newValue) {
			if (response.status == 'error')
				return response.msg;
		}
	});
	
	$('.editable-travelmode').editable({
		type : 'select',
		source: baseURL + '/secure/travel/loadTravelMode.go',
		url : baseURL + '/secure/travel/updatePlans.go',
		success : function(response, newValue) {
			if (response.status == 'error')
				return response.msg;
		}
	});
	
	$('.editable-traveltype').editable({
		type : 'select',
		source: baseURL + '/secure/travel/loadTravelType.go',
		url : baseURL + '/secure/travel/updatePlans.go',
		success : function(response, newValue) {
			if (response.status == 'error')
				return response.msg;
		}
	});

	$('.travel-datepicker').datepicker({
		format : "yyyy-mm-dd",
		weekStart : 0,
		multidate : false,
		autoclose : true,
		todayHighlight: true,
		assumeNearbyYear: true
	});

	/* Full Calendar Plugin */
	var title = '';
	$('#TravelBoard').fullCalendar({
		header : {
			left : 'prev,next today',
			center : 'title',
			right : 'month,basicWeek,basicDay'
		},
		defaultDate : moment().format('YYYY-MM-DD'),
		eventLimit : true, // allow "more" link when too many events
	    events: baseURL + '/secure/travel/planLoad.go',
	    eventMouseover: function(calEvent, jsEvent, view) {
	    	title = $(this).html();
	        $(this).html(getGlyph(calEvent.travelMode)+' ['+calEvent.travelType+'] '+calEvent.title);
	    },
	    eventMouseout: function(calEvent, jsEvent, view) {
	    	$(this).html(title);
	    } 
	});
	
	var getGlyph = function(mode){
		switch (mode) {
		    case 'Motor Bike':
		        return '<i class="fa fa-motorcycle"></i>';
		    case 'Bi Cycle':
		        return '<i class="fa fa-bicycle"></i>';
		    case 'Car':
		    	return '<i class="fa fa-car"></i>';
		    case 'Bus':
		    	return '<i class="fa fa-bus"></i>';
		    case 'Train':
		    	return '<i class="fa fa-train"></i>';
		    case 'Flight':
		    	return '<i class="fa fa-plane"></i>';
		    case 'Ship':
		    	return '<i class="fa fa-ship"></i>';
		    default:
		    	return '';
		}
	}
});